lib/std/mem.zig

```diff
@@ -767,12 +776,12 @@ test "mem.trim" {
 }
 
 /// Linear search for the index of a scalar value inside a slice.
-pub fn indexOfScalar(comptime T: type, slice: []const T, value: T) ?usize {
+pub fn indexOfScalar(comptime T: type, slice: []const T, value: T) ?MaxElements(T) {
     return indexOfScalarPos(T, slice, 0, value);
 }
 
 /// Linear search for the last index of a scalar value inside a slice.
-pub fn lastIndexOfScalar(comptime T: type, slice: []const T, value: T) ?usize {
+pub fn lastIndexOfScalar(comptime T: type, slice: []const T, value: T) ?MaxElements(T) {
     var i: usize = slice.len;
```

Comment:
```suggestion
    var i = @intCast(MaxElements(T), slice.len);
```

would be needed to satisfy the return type. This also applies to the other functions with changed return type which currently want to return a `usize`.

Which begs the question, probably most of the code using these functions will store the results in a `usize` anyway in order to access the index in the slice later on etc. So would this PR make sense in that aspect?